### DISCUSSION TITLE
How do I execute a vscode command from the pygls server?

### DISCUSSION THREAD

**Discussion by andyqtang** [openlawlibrary/pygls#392] [Q&A]
Hi, I'm looking for a way to execute registered commands on the LSP side of things, just like you would on the extension side like so `vscode.commands.executeCommand()`. Is it possible to do this? 
https://code.visualstudio.com/api/extension-guides/command

**Comment by alcarney** [comment]
I don't know of a mechanism in the specification for a server to call an arbitrary command in the client at an arbitrary point in time.

However, it *might* be possible to trigger commands in the client in some situations by filling out fields like [`CodeAction.command`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#codeAction) or [`CompletionItem.command`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem) but I suspect this will vary on a client by client basis.

Another option would be to have your server send a custom request to the client to trigger the execution of a client-side command - but this will require a custom client extension for every client you wish to support.
